include/unifex/get_exception_ptr.hpp

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include <unifex/exception.hpp>
+#include <unifex/tag_invoke.hpp>
+#include <unifex/type_traits.hpp>
+
+#include <memory>
+#include <system_error>
+
+#include <unifex/detail/prologue.hpp>
+
+namespace unifex
+{
+  namespace _get_exception_ptr
+  {
+    inline const struct _fn {
+      // forward std::exception_ptr
+      std::exception_ptr operator()(std::exception_ptr &&eptr) const noexcept {
+        return std::forward<std::exception_ptr>(eptr);
+      }
+
+      // convert std::exception based types to std::exception_ptr
+      template(typename Exception)(
+          requires std::is_base_of_v<std::exception, Exception>)
+          std::exception_ptr
+          operator()(Exception &&ex) const noexcept {
```

Comment:
Formatting.

```suggestion
      template(typename Exception)
          (requires std::is_base_of_v<std::exception, Exception>)
      std::exception_ptr operator()(Exception &&ex) const noexcept {
```

Reply:
Yes, clang-format is a bit confused with this template macro.